### DISCUSSION TITLE
how to build a 2D porous model by your DrawMicromodels

### DISCUSSION THREAD

**Discussion by Jingyuliu20** [GeoChemFoam/GeoChemFoam#7] [Q&A] [answered]
Dear Professor Hannah,

I am using ur Python code to generate a 2D porous code by your DrawMicromodels (https://github.com/hannahmenke/DrawMicromodels)
After running your code, I got the hdf5 files as follows:

perfectlyoffsetcircles_rad60_raddev30_coordev60_repeat0_GIGANTIC.hdf5
...
perfectlyoffsetcircles_rad60_raddev30_coordev60_repeat12_GIGANTIC.hdf5

I am planning to use DBS method, thus i need to get a raw file to get eps field of this image
So how can I convert these hdf5 files into a raw file?

**Comment by GeoChemFoam** [comment]
Hi,

You can open them in fiji and save as raw.

HM

On Dec 18, 2022, at 08:06, Jingyuliu20 ***@***.***> wrote:

﻿
****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************


Dear Professor Hannah,

I am using ur Python code to generate a 2D porous code by your DrawMicromodels (https://github.com/hannahmenke/DrawMicromodels)
After running your code, I got the hdf5 files as follows:

perfectlyoffsetcircles_rad60_raddev30_coordev60_repeat0_GIGANTIC.hdf5
...
perfectlyoffsetcircles_rad60_raddev30_coordev60_repeat12_GIGANTIC.hdf5

I am planning to use DBS method, thus i need to get a raw file to get eps field of this image
So how can I convert these hdf5 files into a raw file?

**Reply by Jingyuliu20** [reply]
Hi Hannah,

I ran your code of Drawmicropmodels without changing any parameter, and translate hdf5 file into raw file format. The raw file is composed of about 12000*12000 points. Is that right? How can i pre-get parameter of future raw file from the Drawmicromodels input parameters. And how can I find the parameter of the raw file. 

And also I cannot compose 13 hdf5 file sequences into one raw file by fiji. Could u give more hints.

**Comment by GeoChemFoam** [comment, marked as answer]
The dimensions are calculated based on the DPI.

You will see the raw parameters when you open the hdf5 in fiji.

The code is not designed to put them in one raw file so you will have to figure out that one on your own.

**Comment by liaoyang123** [comment]
Hi
Have you solved the problem of converting pictures to RAW? I would like to ask you about this question. Thank you.

**Comment by GeoChemFoam** [comment]
Open the HDF5 file in fiji and save them as raw. Alternatively you can edit the script yourself to save as raw – there are plenty of examples of how to do this on stack overflow.


Best wishes,

Hannah

**Comment by liaoyang123** [comment]
By the way, I used HM12_2_2_4 model ( it from:GC-FOAM/image/raw ) in the DBS simulation. But there is a problem of non-convergence, is there any way to solve it?Should I continue to increase the mesh?

**Reply by julienmaes** [reply]
It really depends on what you're trying to do and what's not converging, but refining the mesh is an option

**Reply by liaoyang123** [reply]
Thank you. I plan to use reactiveTransportDBSFoam and try to use this model to simulate dissolution patterns under different Pe and Ki conditions. I've tried refining the grid and this one does work, but the Courant Number get big when I run it for a while. Should I set my injection speed very low? Is there any other better solution?

**Reply by GeoChemFoam** [reply]
Hi Liaoyang, 

Adjusting velocity, mesh resolution, and the timestep are all options for optimising the courant number. Can you track why the courant number is increasing? Maybe the boundary conditions can be changed to constant flow so that the dissolution does not cause an increase in velocity. 

Hannah
